services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/env/EnvCleanupExtension.groovy

```diff
@@ -0,0 +1,155 @@
+package org.openkilda.functionaltests.extension.env
+
+import static org.openkilda.model.MeterId.MAX_SYSTEM_RULE_METER_ID
+
+import org.openkilda.functionaltests.extension.spring.SpringContextExtension
+import org.openkilda.functionaltests.extension.spring.SpringContextListener
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.IslInfoData
+import org.openkilda.messaging.info.event.SwitchInfoData
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.service.database.Database
+import org.openkilda.testing.service.lockkeeper.LockKeeperService
+import org.openkilda.testing.service.northbound.NorthboundService
+import org.openkilda.testing.tools.IslUtils
+
+import groovy.util.logging.Slf4j
+import org.spockframework.runtime.extension.AbstractGlobalExtension
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+
+@Slf4j
+abstract class EnvCleanupExtension extends AbstractGlobalExtension implements SpringContextListener {
+
+    @Autowired
+    TopologyDefinition topology
+
+    @Autowired
+    NorthboundService northbound
+
+    @Autowired
+    Database database
+
+    @Autowired
+    IslUtils islUtils
+
+    @Autowired
+    LockKeeperService lockKeeper
+
+    @Value('${spring.profiles.active}')
+    String profile
+
+    @Override
+    void start() {
+        SpringContextExtension.addListener(this)
+    }
+
+    def deleteAllFlows() {
+        log.info("Deleting all flows")
+        northbound.deleteAllFlows()
+    }
+
+    def unsetLinkMaintenance(List<IslInfoData> links) {
+        def maintenanceLinks = northbound.getAllLinks().findAll { it.underMaintenance }
+        if (maintenanceLinks) {
+            log.info("Unset maintenance mode for affected links: $maintenanceLinks")
+            maintenanceLinks.each {
+                northbound.setLinkMaintenance(islUtils.getLinkUnderMaintenance(it, false, false))
+            }
+        }
+    }
+
+    def reviveFailedLinks(List<IslInfoData> links) {
+        def failedLinks = links.find { it.state == IslChangeType.FAILED }
+        if (failedLinks) {
+            log.info("Bring ports up for failed links: $failedLinks")
+            failedLinks.each {
+                try {
+                    northbound.portUp(it.source.switchId, it.source.portNo)
+                    northbound.portUp(it.destination.switchId, it.destination.portNo)
+                } catch (Throwable t) {
+                    // Switch may be disconnected. Ignore for now, healthcheck test will report problems if any.
+                    // This situation may require manual intervention
+                }
+            }
+        }
+    }
+
+    def deleteLinkProps() {
+        def linkProps = northbound.getAllLinkProps()
+        if (linkProps) {
+            log.info("Deleting all link props")
```

Comment:
According to your answer above, let's log link props as well. I like the idea to be more verbose.

Reply:
done